internal/internal_command_state_machine.go

```diff
@@ -284,6 +293,65 @@ func (h *commandsHelper) newCommandStateMachineBase(commandType commandType, id
 	}
 }
 
+func (h *commandsHelper) acceptWorkflowUpdate(updateID string) {
```

Comment:
While we don't _necessarily_ need to do it here, often their pattern was to create one of these command state machines for an update, and then have several methods on the _same_ state machine. So a single update state machine could have `accept`, `complete`, `reject`, etc.

This can help with cancellation which may not matter here, but also may keep things a bit cleaner. A multi-interaction thing like an update may be clearer as a single state machine with actions that can be called on it.

Reply:
I think I'm misunderstanding the commandStateMachine abstraction. I had believed it to be a SM _for commands_. Like, every single command gets an appropriate SM instance. Accept/Complete being distinct commands would thus be distinct SM types. But it seems like you're saying that these command SMs are really more general and are intended to be used throughout event processing?

Reply:
> I had believed it to be a SM _for commands_

Kinda. It's basically for every "state" a "thing" can go through. The result is commands, but it's really for "things", which can result in multiple states which we don't have many of. Currently:

* An activity state machine can send schedule or cancel (though this state machine starts a cancel state machine for unknown reasons rather than sending the cancel itself)
* A child workflow state machine that can send a start or cancel
* A timer state machine that can send a start or cancel (though like activity, this itself starts a separate cancel state machine which I think is silly)

This will be the first "thing" that can have three commands that I know of. It can accept (leaving it around), complete (removing itself), and reject (removing itself).